src/irrigation_station.py

```diff
@@ -9,22 +10,42 @@ class IrrigationPoint:
     STATE_OPEN = "open"
     STATE_CLOSED = "closed"
 
-    def __init__(self, config: IrrigationPointConfig, logger: Logger) -> None:
+    def __init__(self, config: IrrigationPointConfig, ads_modules: dict[int, ADS1115], logger: Logger) -> None:
         """Initialize an irrigation point with its configuration."""
         self.config = config
         self._valve_state: str = IrrigationPoint.STATE_CLOSED
-        self._sensor = ADC(config.sensor_pin)
         self._valve = Pin(config.valve_pin, Pin.OUT)
+        self._mosfet = Pin(config.mosfet_pin, Pin.OUT)
+        self._ads = ads_modules[config.ads_address]
         self._sensor_value = 0.5
         self._logger = logger
+        
+        # Ensure sensor is powered off initially
+        self._mosfet.value(0)
 
     def get_sensor_value(self) -> float:
-        """Simulate and return the current soil moisture sensor value (0.0-1.0)."""
-        conversion_factor = 1 / (65535)
-        self._sensor_value = self._sensor.read_u16() * conversion_factor
-        # self._logger.log(
-        #     f"[Sensor] {self.config.name}: Moisture level is {self._sensor_value * 100:.1f}%"
-        # )
+        """Read the current soil moisture sensor value (0.0-1.0) using ADS1115."""
+        # Power on the sensor
+        self._mosfet.value(1)
+        
+        # Wait for sensor to stabilize
+        sleep_ms(100)
+        
+        try:
+            # Read from ADS1115
+            raw_value = self._ads.read(self.config.ads_channel)
+            
+            # Convert to 0.0-1.0 range (ADS1115 is 16-bit signed: -32768 to 32767)
+            # Assuming single-ended measurement, values should be 0 to 32767
+            self._sensor_value = max(0.0, min(1.0, raw_value / 32767.0))
```

Comment:
I am concerned about the min/max calls here. While this does ensure a correct value is computed all the time, I think this could also mask issues. I think a different approach would be better...

Let's implement as follows:
- wrap in a try/catch
- compute a 0-1 value based on expected readings
- if the computation itself does not throw, but an invalid value is computed, throw an error too
- in the catch block, use the logger to write relevant info to the log